ocrd/ocrd/cli/resmgr.py

```diff
@@ -45,7 +45,11 @@ def list_available(executable=None):
     List available resources
     """
     resmgr = OcrdResourceManager()
-    for executable, reslist in resmgr.list_available(executable):
+    resources = resmgr.list_available(executable)
+    if executable and not resources:
+        print("Error: No resources available for '%s'" % executable)
+        sys.exit(1)
+    for executable, reslist in resources:
```

Comment:
That hunk should be reverted for several reasons:
- we cannot just `print`, that would mix expected output (usually to be parsed by another tool) with messages (which should use logging and thus end up in a file or on stderr)
- we don't want a failure, just an empty result in that case
- it's not necessary, your fix in `OcrdResourceManager.list_available` is already sufficient

```suggestion
    for executable, reslist in resmgr.list_available(executable):
```

Reply:
Yes, I understand the @joschrew's intention to give a better error messages. But I agree with @bertsky this is not the right place, or rather it is not necessary.

If we want to make it explicit that there are no resources for an executable, the right place to adapt ist the `print_resources` function.